client/src/lib/query/use-task-mutation.ts

```diff
@@ -0,0 +1,23 @@
+import { createRequestConfig } from "@/lib/fetch/create-request-config";
+import { makeAuthorizedUrl } from "@/lib/fetch/make-authorized-url";
+import { queryClient } from "@/lib/query-client";
+import { Data } from "@/types/query.types";
+import { Activity, ActivityUpdateInput } from "@/types/server/activity.types";
+import { useMutation } from "@tanstack/react-query";
+
+async function putTaskCompletion(input: ActivityUpdateInput) {
```

Comment:
Time to abstract this logic away maybe. Call it `request.put()` or something. Most of them just build the URL and put an object in the request body and I've already noticed it's easy to forget to `makeAuthorizedUrl()`.

Reply:
See #31. Not going to prioritize this right now.